R/surv.b.R

```diff
@@ -40,150 +52,237 @@ survClass <- R6::R6Class(
                         row <- list()
+                        row[[paste0('chisqr[', test, ']')]] <- DF$chisqr[i]
+                        row[[paste0('df[', test, ']')]] <- DF$df[i]
+                        row[[paste0('pvalue[', test, ']')]] <- DF$pvalue[i]
                         
-                        row[[paste0('z[', test, ']')]] <- result$statistic['z']
-                        row[[paste0('nu[', test, ']')]] <- result$statistic['nu']
-                        row[[paste0('nuse[', test, ']')]] <- sqrt(result$statistic['var.nu'])
-                        row[[paste0('p[', test, ']')]] <- result$p.value
-                        
-                        tt$setRow(rowKey=pair, values=row)
+                        tt$setRow(rowKey=1, values=row)
                     }
+                    
                 }
+                tt$setNote("note1","Log-Rank and Peto-Peto tests use survdiff from the survival package. Gehan and Tarone-Ware tests are obtained from the coin package.",init=TRUE)
+
             }
+
+            
+            
+            
+
+            
         },
-        .plot=function(image, theme, ggtheme, ...) {
+        .plot=function(image, ...) {
             
-            state <- image$state
-            if (is.null(state))
-                return(FALSE)
+            eventVarName <- self$options$event
+            elapsedVarName <- self$options$elapsed
             
-            if (identical(image, self$results$sc)) {
-                ylab <- 'Survival'
-                fun <- NULL
-                cens <- self$options$cens
-                ci <- self$options$ci
-                ylim <- c(0, 1)
-            } else if (identical(image, self$results$hf)) {
-                
-                return(FALSE)
-                
-                ylab <- 'Hazard Function'
-                fun <- 'event'
-                cens <- FALSE
-                ci <- FALSE
-                ylim <- c(0, 1)
-            } else {
-                ylab <- 'Cumulative Hazard'
-                fun <- 'cumhaz'
-                cens <- FALSE
-                ci <- FALSE
-                ylim <- NULL
-            }
+            if (is.null(eventVarName) || is.null(elapsedVarName))
+                return()
+            
+            tmpDat <- image$state 
             
-            plot <- ggfortify:::autoplot.survfit(
-                object=state,
-                fun=fun,
-                xlab='Elapsed',
-                ylab=ylab,
-                ylim=ylim,
-                surv.size = 1,
-                censor.size = 8,
-                censor.alpha = 0.8,
-                conf.int=ci,
-                censor=cens)
+            s <- survival::survfit(survival::Surv(times, status) ~ group, data=tmpDat)
+            if (!is.null(names(s$strata))) names(s$strata) <- gsub("group=", "", names(s$strata))
+
+            conf.int <- self$options$ci
+            censYN <- self$options$cens
+            risktableYN <- self$options$risktable
```

Comment:
i'd discourage the use of `.`s in variable names (in death watch i've used camelCase), and i'd discourage the `YN` suffix ... but i probably don't care that much in the end.

Reply:
Ok. I changed the names of the objects to 'ci', 'cens' and 'risk'